### PR TITLE
Correct spelling errors throughout the project

### PR DESCRIPTION
**Pull request by kupolak**
As mentioned in the title - address typos.

### CODE REVIEW

File: vendor/0install-solver/src/solver/diagnostics.ml
Comment (emillon) [inline vendor/0install-solver/src/solver/diagnostics.ml:1]: source code in the `vendor/` directory is not being developed here (it's pulled from other projects), so we can't fix it this way. can you remove this part from this PR?

**Review by emillon — COMMENTED** [review]
Thanks a lot. I checked the changes and it's only typos that can be changed without complications. But the `vendor/` parts cannot be fixed this way.

File: vendor/0install-solver/src/solver/diagnostics.ml
Comment (kupolak) [reply vendor/0install-solver/src/solver/diagnostics.ml:1]: Sure, I removed it.
Does it make sense to use git submodules for these other projects? That could be a potential improvement I suppose but I'm not sure if it's doable.

Comment (emillon) [reply vendor/0install-solver/src/solver/diagnostics.ml:1]: thanks for the suggestion, but it's not directly mapping to an existing repository so we don't use submodules in dune.

**Pull request merged**